### PR TITLE
Add SWIFT_PLATFORMS_USING_ISYSROOT to list platforms that should be building with -isysroot

### PR DESCRIPTION
**Pull request by kubamracek**

### CODE REVIEW

**Comment by compnerd** [comment]
Hmm, I'm not sure I understand this.  If I'm working with freestanding, I could be on a Linux or windows environment, which don't use `isysroot`.  I realize that those configurations aren't in CI currently, but this configuration makes that more challenging.  Maybe we should have a prefer isysroot option instead that's false and true on Darwin hosts by default but the user can change?

**Comment by kubamracek** [comment]
@compnerd Updated. Notice that currently the FREESTANDING SDK is set up as `configure_sdk_darwin`, so it does get USE_ISYSROOT=1. If we wanted to actually start building FREESTANDING on Linux, we'd have to add a `configure_sdk_unix` somewhere.

**Comment by kubamracek** [comment]
@swift-ci please test

File: cmake/modules/SwiftConfigureSDK.cmake [hunk 1]
```diff
@@ -145,6 +145,8 @@ endfunction()
 #   SWIFT_SDK_${prefix}_LIB_SUBDIR          Library subdir for this SDK
 #   SWIFT_SDK_${prefix}_VERSION_MIN_NAME    Version min name for this SDK
 #   SWIFT_SDK_${prefix}_TRIPLE_NAME         Triple name for this SDK
+#   SWIFT_SDK_${prefix}_OBJECT_FORMAT       The object file format (e.g. MACHO)
```
Comment (compnerd) [inline cmake/modules/SwiftConfigureSDK.cmake:148]: Hah!  I think you put this in the wrong commit 🤷‍♂️

**Review by compnerd — APPROVED** [review]

File: cmake/modules/SwiftConfigureSDK.cmake (hunk 1 above)
Comment (kubamracek) [reply cmake/modules/SwiftConfigureSDK.cmake:148]: No, this was an intentional drive-by addition :) It seemed to be missing...

**Pull request merged**